File: bubble_sort/js/bubbleSort.js
const swap = (arr, i, j) => {
  let temp = arr[i];
  arr[i] = arr[j];
  arr[j] = temp;
}

const bubbleSort = (arr) => {
  let i = 0;
  while (i < arr.length - 1) {
    if (arr[i] > arr[i + 1]) {
      swap(arr, i, i + 1);
      i = 0;
    } else {
      i++;
    }
  }
  return arr;
}

console.log(bubbleSort([2, 1, 3, 4]));

File: bubble_sort/python/bubbleSort.py
def swap(arr, i, j):
  temp = arr[i]
  arr[i] = arr[j]
  arr[j] = temp

def bubbleSort(arr):
  unsorted = True
  while unsorted:
    unsorted = False
    for i in range(len(arr) - 1):
      if arr[i] > arr[i + 1]:
        swap(arr, i, i + 1)
        unsorted = True
  return arr

print(bubbleSort([2, 1, 3, 4]))